language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/codeaction/AddModuleToBallerinaTomlCodeActionTest.java

```diff
@@ -0,0 +1,132 @@
+import java.util.stream.Collectors;
+
+/**
+ * Test class to test module addition to Ballerina.toml.
+ *
+ * @since 2201.8.0
+ */
+public class AddModuleToBallerinaTomlCodeActionTest extends AbstractCodeActionTest {
+    
+    protected void setupLanguageServer(TestUtil.LanguageServerBuilder builder) {
+        builder.withInitOption(InitializationOptions.KEY_POSITIONAL_RENAME_SUPPORT, true);
+    }
+
+    @BeforeClass
+    @Override
+    public void setup() {
+        super.setup();
+
+        LanguageServerContext context = new LanguageServerContextImpl();
+        BallerinaLanguageServer languageServer = new BallerinaLanguageServer();
+        Endpoint endpoint = TestUtil.initializeLanguageSever(languageServer);
+        try {
+            Map<String, String> localProjects = Map.of("pkg1", "main.bal", "pkg2", "main.bal", "x", "main.bal", 
+                    "x.y", "main.bal");
+            List<LSPackageLoader.ModuleInfo> localPackages = getLocalPackages(localProjects,
+                    languageServer.getWorkspaceManager(), context).stream().map(LSPackageLoader.ModuleInfo::new)
+                    .collect(Collectors.toList());
+            Mockito.when(getLSPackageLoader().getLocalRepoPackages(Mockito.any())).thenReturn(localPackages);
+        } catch (Exception e) {
+            //ignore
+        } finally {
+            TestUtil.shutdownLanguageServer(endpoint);
+        }
+    }
+
+    @Test(dataProvider = "codeaction-data-provider")
+    @Override
+    public void test(String config) throws IOException, WorkspaceDocumentException {
+        super.test(config);
+    }
+
+    @DataProvider(name = "codeaction-data-provider")
+    @Override
+    public Object[][] dataProvider() {
+        return new Object[][]{
+                {"add_module1.json"},
+                {"add_module2.json"},
+                {"add_module3.json"},
+                {"add_module4.json"},
+                {"add_module5.json"},
+                {"add_module6.json"},
+                {"add_module7.json"},
+                {"add_module8.json"},
+                {"add_module9.json"},
+                {"add_module10.json"},
+        };
+    }
+
+    @Test(dataProvider = "negativeDataProvider")
+    @Override
+    public void negativeTest(String config) throws IOException, WorkspaceDocumentException {
+        super.negativeTest(config);
+    }
+
+    @DataProvider
+    public Object[][] negativeDataProvider() {
+        return new Object[][]{
+                {"add_module_negative_1.json"},
+                {"add_module_negative_2.json"}
+        };
+    }
+
+    @Override
+    public String getResourceDir() {
+        return "add-module-ballerina-toml";
+    }
+
+    @Override
+    public boolean loadMockedPackages() {
+        return true;
+    }
+
+    private static List<Package> getLocalPackages(Map<String, String> projects, WorkspaceManager workspaceManager,
+                                                    LanguageServerContext context)
```

Comment:
remove extra white spaces before LanguageServerContext to align it with the first parameter.